src/core/fieldformatter/qgsvaluerelationfieldformatter.cpp

```diff
@@ -0,0 +1,149 @@
+/***************************************************************************
+  qgsvaluerelationfieldformatter.cpp - QgsValueRelationFieldFormatter
+
+ ---------------------
+ begin                : 3.12.2016
+ copyright            : (C) 2016 by Matthias Kuhn
+ email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsvaluerelationfieldformatter.h"
+
+#include "qgis.h"
+#include "qgsmaplayerregistry.h"
+#include "qgsvectorlayer.h"
+
+#include <QSettings>
+
+bool orderByKeyLessThan( const QgsValueRelationFieldFormatter::ValueRelationItem& p1, const QgsValueRelationFieldFormatter::ValueRelationItem& p2 )
+{
+  return qgsVariantLessThan( p1.key, p2.key );
+}
+
+bool orderByValueLessThan( const QgsValueRelationFieldFormatter::ValueRelationItem& p1, const QgsValueRelationFieldFormatter::ValueRelationItem& p2 )
+{
+  return qgsVariantLessThan( p1.value, p2.value );
+}
+
+
+QgsValueRelationFieldFormatter::QgsValueRelationFieldFormatter()
+{
+
+}
+
+QString QgsValueRelationFieldFormatter::id() const
+{
+  return QStringLiteral( "ValueRelation" );
+}
+
+QString QgsValueRelationFieldFormatter::representValue( QgsVectorLayer* layer, int fieldIndex, const QVariantMap& config, const QVariant& cache, const QVariant& value ) const
+{
+  Q_UNUSED( layer )
+  Q_UNUSED( fieldIndex )
+
+  QgsValueRelationFieldFormatter::ValueRelationCache vrCache;
```

Comment:
Since you are inside a method of `QgsValueRelationFieldFormatter`, you can remove the `QgsValueRelationFieldFormatter::`